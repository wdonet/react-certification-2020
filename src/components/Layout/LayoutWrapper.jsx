import React, { useContext } from 'react';
import styled, { ThemeContext } from 'styled-components';
import Header from './Header';
import HomeView from './HomeView';
import { SearchWrapper } from '../../wrappers/SearchWrapper';

const StyledLayoutWrapper = styled.div`
  background: ${({ theme }) => theme.color.background};
  width: 100%;
  height: 100%;
`;

const LayoutWrapper = ({ children }) => {
  const { theme } = useContext(ThemeContext);

  return (
    <StyledLayoutWrapper data-testid="layout-wrapper" theme={theme} role="application">
      <SearchWrapper>
        <Header />
        <HomeView>{children}</HomeView>
      </SearchWrapper>
    </StyledLayoutWrapper>
  );
};

export default LayoutWrapper;
