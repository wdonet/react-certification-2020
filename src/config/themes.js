export default {
  dark: {
    icon: 'moon',
    sound: 'https://www.joshwcomeau.com/sounds/switch-on.mp3',
    background: {
      primary: '#161f23',
      secondary: '#1e2931',
    },
    decorator: {
      primary: '#2c0e7b',
      secondary: '#194c7c',
      contrast: '#FFF',
    },
    text: {
      primary: '#FFF',
      secondary: '#FFF',
    },
  },
  light: {
    icon: 'sun',
    sound: 'https://www.joshwcomeau.com/sounds/switch-off.mp3',
    background: {
      primary: '#eef4f4',
      secondary: '#d7dee7',
    },
    decorator: {
      primary: '#5b8cbd',
      secondary: '#d29b3b',
      contrast: '#222',
    },
    text: {
      primary: '#222',
      secondary: '#222',
    },
  },
}
