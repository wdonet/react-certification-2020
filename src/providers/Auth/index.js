export { default, AuthContext } from './AuthProvider';
export { default as useAuth } from './useAuth';
