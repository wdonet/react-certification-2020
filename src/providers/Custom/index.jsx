import React, { useState } from 'react'
import PropTypes from 'prop-types'
import customs from '../../config/customs'

export const CustomContext = React.createContext({})

const CustomProvider = ({ children }) => {
  const [triggered, setTriggered] = useState({})
  const [themeClicks, setThemeClicks] = useState(0)

  if (themeClicks >= 10) {
    setTriggered({
      ...triggered,
      rainbow: {
        values: customs.rainbow,
        activated: true,
      },
    })
    setThemeClicks(0)
  }

  const increaseThemeClicks = () => {
    setThemeClicks(themeClicks + 1)
  }

  const contextValue = {
    triggered,
    increaseThemeClicks,
  }

  return <CustomContext.Provider value={contextValue}>{children}</CustomContext.Provider>
}

CustomProvider.propTypes = {
  children: PropTypes.node.isRequired,
}

export default CustomProvider
