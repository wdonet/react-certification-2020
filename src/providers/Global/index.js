export { default, GlobalContext } from './GlobalProvider';
export { default as useGlobal } from './useGlobal';
