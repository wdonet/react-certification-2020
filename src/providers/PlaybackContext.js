import React from 'react';

const PlaybackContext = React.createContext({ setVideoID: null });

export default PlaybackContext;
