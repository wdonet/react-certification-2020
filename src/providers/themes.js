const defaultColors = {
  white: '#FFFFFF',
  black: '#000000',
};

const font = {
  h1: {
    fontSize: '',
    fontWeight: '',
    letterSpacing: 1.5,
  },
};

export const lightTheme = {
  font,
  color: {
    ...defaultColors,

    primary: '#dce775',
    primaryVariant: '#c0ca33',
    secondary: '#c8ff00',

    background: '#ffffff',
    surface: '#f5f5f5',
    error: '#e64a19',
  },
};

export const darkTheme = {
  font,
  color: {
    ...defaultColors,

    primary: '#00695c',
    primaryVariant: '#004d40',
    secondary: '#1de9b6',

    background: '#616161',
    surface: '#757575',
    error: '#e65100',
  },
};
