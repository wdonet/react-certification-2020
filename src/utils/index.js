export { default as contextWrapper } from './testing/helpers/contextWrapper';
export { default as renderWithTheme } from './testing/helpers/renderWithTheme';
export { default as googleMockedAPIObject } from './testing/mocks/googleMockedAPIObject';
export { default as youtubeMockedData } from './testing/mocks/youtubeMockedData';
export { default as YTMockedObject } from './testing/mocks/YTMockedObject';
