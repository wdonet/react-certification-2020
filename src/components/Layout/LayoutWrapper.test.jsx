import React from 'react';
import 'jest-styled-components';
import { getByRole, render, act } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { googleMockedAPIObject } from '../../utils';
import { lightTheme } from '../../providers/themes';
import LayoutWrapper from './LayoutWrapper';

global.gapi = googleMockedAPIObject();

const build = async (Component = <LayoutWrapper />) => {
  let container;
  await act(async () => {
    container = render(
      <ThemeProvider theme={{ theme: lightTheme }}>{Component}</ThemeProvider>
    ).container;
  });
  return {
    container,
    Header: () => getByRole(container, 'toolbar'),
    HomeView: () => getByRole(container, 'main'),
  };
};

describe('LayoutWrapper', () => {
  it('renders', async () => {
    const wrapper = await build();
    expect(wrapper).toMatchSnapshot();
  });

  it('displays Header and HomeView', async () => {
    const { Header, HomeView } = await build();

    expect(Header()).toBeInTheDocument();
    expect(HomeView()).toBeInTheDocument();
  });
});

describe('LayoutWrapper styles', () => {
  it('applies default styling', async () => {
    const { firstChild } = (await build()).container;
    expect(firstChild).toHaveStyle('width: 100%');
    expect(firstChild).toHaveStyle('height: 100%');
  });
});
