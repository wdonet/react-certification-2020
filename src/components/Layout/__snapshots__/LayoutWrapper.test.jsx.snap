// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutWrapper renders 1`] = `
Object {
  "Header": [Function],
  "HomeView": [Function],
  "container": <div>
    <div
      class="sc-kpOJdX cseeBf"
      data-testid="layout-wrapper"
      role="application"
    >
      <div
        class="sc-chPdSV bcCQhf"
        data-testid="header"
        role="toolbar"
      >
        <div
          class="sc-gZMcBi dqmLfP"
        >
          <div
            class="sc-jzJRlG lfkGpG"
            data-testid="sidebar"
          >
            <div
              class="sc-cSHVUG cTUJGP"
            >
              Welcome
              <img
                alt="close"
                class="sc-gzVnrw iCjpdg"
                role="button"
                src="close_icon.png"
              />
            </div>
            <div
              class="sc-kAzzGY dLFxrJ"
            >
              <button
                class="sc-iwsKbI dHHiSG"
                data-testid="sidebar-home"
                height="30px"
                width="calc( 100% - 8px )"
              >
                Home
              </button>
              <button
                class="sc-iwsKbI dHHiSG"
                data-testid="sidebar-favorites"
                height="30px"
                width="calc( 100% - 8px )"
              >
                Favorites
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-gZMcBi dqmLfP"
        >
          <div
            class="sc-gqjmRU fsrssQ"
            data-testid="login-form"
            role="form"
          >
            <div
              class="sc-VigVT lsdIp"
            >
              Login!
            </div>
            <div
              class="sc-VigVT lsdIp"
            >
              <label
                for="username"
              >
                User
              </label>
              <input
                class="sc-dnqmqq eMiNCI"
                data-testid="username-input"
                id="username"
                name="username"
              />
            </div>
            <div
              class="sc-VigVT lsdIp"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                class="sc-dnqmqq eMiNCI"
                data-testid="password-input"
                id="password"
                name="password"
                type="password"
              />
            </div>
            <div
              class="sc-VigVT lsdIp"
            >
              <span
                style="color: red;"
              />
            </div>
            <div
              class="sc-VigVT gRVnnS"
              display="flex"
            >
              <button
                class="sc-iwsKbI ffUqSO"
                data-testid="login-button"
                height="30px"
              >
                Login
              </button>
              <button
                class="sc-iwsKbI ffUqSO"
                data-testid="cancel-button"
                height="30px"
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-kgoBCf sZeiO"
        >
          <img
            alt="hamburguer"
            class="sc-gzVnrw iCjpdg"
            role="button"
            src="hamburguer_icon.jpeg"
          />
          <input
            class="sc-dnqmqq eMiNCI"
            data-testid="search-input"
            role="search"
          />
        </div>
        <div
          class="sc-kgoBCf sZeiO"
        >
          <input
            class="sc-htoDjs bZzhDu"
            data-testid="theme-mode-switch"
            role="switch"
            type="checkbox"
          />
          <div
            class="sc-jTzLTM cqwcFw"
          >
            <div
              class="menu-button"
            >
              <img
                alt="profile"
                class="sc-bdVaJa ieaWl"
                data-testid="user-avatar"
                role="button"
                src="default_user.jpg"
              />
            </div>
            <div
              class="sc-fjdhpX gixdjM"
              data-testid="dropdown-content"
            >
              <button
                class="sc-iwsKbI djEUtr"
                data-testid="menu-login-button"
                height="30px"
                width="100%"
              >
                Login
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-kGXeez cYLHhW"
        data-testid="main-content"
        role="main"
      />
    </div>
  </div>,
}
`;
